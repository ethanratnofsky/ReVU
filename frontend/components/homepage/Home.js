import { FlatList, Image, SafeAreaView, Text, TouchableOpacity, View } from 'react-native';

import homeStyles from './HomeStyles';

import DiningHallButton from './DiningHallButton';
import Button from '../button/Button';

import { DINING_HALLS, ANONYMOUS_NAMES } from '../../constants';

const Home = ({ navigation, route }) => {
    const currDate = new Date();
    const weekday = new Intl.DateTimeFormat('en-US', { weekday: 'long' }).format(currDate);
    const month = new Intl.DateTimeFormat('en-US', { month: 'long' }).format(currDate);
    const date = currDate.getDate();
    const year = currDate.getFullYear();
    const uid = route.params.id;

    const renderDiningHallButton = ({item}) => (
        <DiningHallButton id={item.id} name={item.name} userId={uid}/>
    );

    return (
        <SafeAreaView style={homeStyles.container}>
            <View style={homeStyles.header}>
                <Image style={homeStyles.logo} source={require('../../assets/images/revu-logo.png')} />
                <TouchableOpacity style={homeStyles.userContainer} onPress={() => alert("user")}>
                    <Text style={homeStyles.userName}>User</Text>
                    <Image style={homeStyles.userIcon} source={require('../../assets/images/white-user.png')} />
                </TouchableOpacity>
            </View>
            <View style={homeStyles.contentContainer}>
                <Text style={homeStyles.date}>{weekday}, {month} {date}, {year}</Text>
                <FlatList 
                    data={DINING_HALLS} 
                    renderItem={renderDiningHallButton} 
                    keyExtractor={item => item.id}
                    scrollEnabled={false}
                />
            </View>
            <View style={homeStyles.footer}>
                <TouchableOpacity style={homeStyles.termsContainer} onPress={() => navigation.navigate('Terms and Conditions', { firstTime: false, id: uid })}>
                    <Image style={homeStyles.termsIcon} source={require('../../assets/images/white-terms.png')} />
                    <Text style={homeStyles.termsText}>Terms & Conditions</Text>
                </TouchableOpacity>
                <Button fontSize={12} imgSrc={require('../../assets/images/gold-complaints.png')} onPress={() => navigation.navigate("Complaints")} style={homeStyles.complaintsButton} text='File a Complaint' />
            </View>
        </SafeAreaView>
    );
}

export default Home;
