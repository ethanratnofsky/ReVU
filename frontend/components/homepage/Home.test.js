import React from 'react';
const renderer = require('react-test-renderer');
import Home from './Home';

describe('<Home />', () => {
  it('has 3 children', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree.children.length).toBe(3);
  });
});

it('renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });