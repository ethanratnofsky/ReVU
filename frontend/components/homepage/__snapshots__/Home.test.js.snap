// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1C1C1C",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": "10%",
        "marginTop": 40,
        "width": "90%",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 39,
          "width": 134,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        User
      </Text>
      <Image
        source={1}
        style={
          Object {
            "height": 24,
            "marginLeft": 10,
            "width": 24,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "LibreCaslonText_400Regular",
          "fontSize": 16,
          "marginVertical": "5%",
        }
      }
    >
      Wednesday
      , 
      November
       
      16
      , 
      2022
    </Text>
    <RCTScrollView
      data={
        Array [
          Object {
            "id": 0,
            "name": "Rand Dining Hall",
            "overallRating": 5,
          },
          Object {
            "id": 1,
            "name": "Kissam Dining Hall",
            "overallRating": 4.3,
          },
          Object {
            "id": 2,
            "name": "Rothschild Dining Hall",
            "overallRating": 3,
          },
          Object {
            "id": 3,
            "name": "E. B. I. Dining Hall",
            "overallRating": 4.2,
          },
          Object {
            "id": 4,
            "name": "Zeppos Dining Hall",
            "overallRating": 2.5,
          },
          Object {
            "id": 5,
            "name": "Commons Dining Hall",
            "overallRating": null,
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEnabled={false}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#CFAE70",
                "borderRadius": 6,
                "borderStyle": "solid",
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 55,
                "justifyContent": "space-between",
                "margin": 10,
                "opacity": 1,
                "padding": 15,
                "width": 300,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1C1C1C",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Rand Dining Hall
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1C1C1C",
                    "fontFamily": "LibreCaslonText_400Regular",
                    "fontSize": 16,
                  }
                }
              >
                NaN
              </Text>
              <Image
                source={1}
                style={
                  Object {
                    "aspectRatio": 1,
                    "bottom": 1,
                    "left": 1,
                    "position": "relative",
                    "width": 24,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#CFAE70",
                "borderRadius": 6,
                "borderStyle": "solid",
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 55,
                "justifyContent": "space-between",
                "margin": 10,
                "opacity": 1,
                "padding": 15,
                "width": 300,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1C1C1C",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Rand Dining Hall
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1C1C1C",
                    "fontFamily": "LibreCaslonText_400Regular",
                    "fontSize": 16,
                  }
                }
              >
                NaN
              </Text>
              <Image
                source={1}
                style={
                  Object {
                    "aspectRatio": 1,
                    "bottom": 1,
                    "left": 1,
                    "position": "relative",
                    "width": 24,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#CFAE70",
                "borderRadius": 6,
                "borderStyle": "solid",
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 55,
                "justifyContent": "space-between",
                "margin": 10,
                "opacity": 1,
                "padding": 15,
                "width": 300,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1C1C1C",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Rand Dining Hall
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1C1C1C",
                    "fontFamily": "LibreCaslonText_400Regular",
                    "fontSize": 16,
                  }
                }
              >
                NaN
              </Text>
              <Image
                source={1}
                style={
                  Object {
                    "aspectRatio": 1,
                    "bottom": 1,
                    "left": 1,
                    "position": "relative",
                    "width": 24,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#CFAE70",
                "borderRadius": 6,
                "borderStyle": "solid",
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 55,
                "justifyContent": "space-between",
                "margin": 10,
                "opacity": 1,
                "padding": 15,
                "width": 300,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1C1C1C",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Rand Dining Hall
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1C1C1C",
                    "fontFamily": "LibreCaslonText_400Regular",
                    "fontSize": 16,
                  }
                }
              >
                NaN
              </Text>
              <Image
                source={1}
                style={
                  Object {
                    "aspectRatio": 1,
                    "bottom": 1,
                    "left": 1,
                    "position": "relative",
                    "width": 24,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#CFAE70",
                "borderRadius": 6,
                "borderStyle": "solid",
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 55,
                "justifyContent": "space-between",
                "margin": 10,
                "opacity": 1,
                "padding": 15,
                "width": 300,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1C1C1C",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Rand Dining Hall
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1C1C1C",
                    "fontFamily": "LibreCaslonText_400Regular",
                    "fontSize": 16,
                  }
                }
              >
                NaN
              </Text>
              <Image
                source={1}
                style={
                  Object {
                    "aspectRatio": 1,
                    "bottom": 1,
                    "left": 1,
                    "position": "relative",
                    "width": 24,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#CFAE70",
                "borderRadius": 6,
                "borderStyle": "solid",
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 55,
                "justifyContent": "space-between",
                "margin": 10,
                "opacity": 1,
                "padding": 15,
                "width": 300,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#1C1C1C",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Rand Dining Hall
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1C1C1C",
                    "fontFamily": "LibreCaslonText_400Regular",
                    "fontSize": 16,
                  }
                }
              >
                NaN
              </Text>
              <Image
                source={1}
                style={
                  Object {
                    "aspectRatio": 1,
                    "bottom": 1,
                    "left": 1,
                    "position": "relative",
                    "width": 24,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 20,
        "width": "90%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginRight": 2,
            "width": 20,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 12,
            "textDecorationLine": "underline",
          }
        }
      >
        Terms & Conditions
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 40,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4294897334,
              4290024008,
            ],
            "endPoint": Array [
              1,
              0,
            ],
            "locations": undefined,
            "startPoint": Array [
              0,
              1,
            ],
          }
        }
        style={
          Object {
            "borderRadius": 10,
            "padding": 3,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 8,
              "flexDirection": "row",
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 16,
                "marginRight": 10,
                "width": 12,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#CFAE70",
                "fontSize": 12,
                "fontWeight": "bold",
              }
            }
          >
            File a Complaint
          </Text>
        </View>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
</RCTSafeAreaView>
`;
